### ISSUE TITLE
Add support for :not selector (and other pseudo-class selectors)

### ISSUE THREAD

**Issue by jjcarstens** [philss/floki#82]
In my case, I'm attempting to use the pseudo-class `:not`selector as `"div:not(.invisible)#error"`

```elixir
pry(1)> Floki.find(html, "div:not(.invisible)#error")
iex(15)> [warn] Unknown token '('. Ignoring.
[warn] Unknown token ')'. Ignoring.
[warn] Pseudo-class "not" is not implemented. Ignoring.
[]
```

I would loooovvee to have support for `:not`

**Comment by philss** [comment]
Hi @jjcarstens!

This feature would be really cool to have!

I think the correct syntax for your selector would be `div#error:not(.invisible)`.

For the implementation we would probably change a little bit the [selector parser](https://github.com/philss/floki/blob/master/lib/floki/selector_parser.ex) and perhaps add some rules or change [the lexer](https://github.com/philss/floki/blob/master/src/floki_selector_lexer.xrl) to consider a simple selector inside the `:not` parenthesis.

Here is the W3C specs for `:not()` pseudo-class: https://drafts.csswg.org/selectors-3/#negation

I'm gonna let this open for a while until I can work on this. Feel free to give a try if you want too.

Thanks!

**Comment by jjcarstens** [comment]
Thanks for the tip! That gets me pointed in the right direction so if I find some time I can take a gander as well.

Looks like both our selectors work (at least testing within the console), but yours definitely reads easier.

Here's a quick test case that I've been using (in case you need it)
```html
<!-- Test selector: div#error:not(.invisible) -->
<div class="foo" id="error">Should match!</div>
<div class="bar invisible" id="error">..No match..</div>
```

**Comment by Eiji7** [comment]
More info at [MDN](https://developer.mozilla.org/en-US/docs/Web/CSS/Pseudo-classes), [Selectors Level 4 Editor’s Draft, 27 January 2017.](https://drafts.csswg.org/selectors/) and [Category: Selectors at jQuery docs](https://api.jquery.com/category/selectors/).
Maybe also some `Floki` and / or jQuery special selectors like: `:floki-has-text('     a text       ')` and `:floki-has-trimmed-text("a text")`?

**Comment by philss** [comment]
@Eiji7 thank you for the links! I think we could have a separated issue for this suggestion you made. Can you create that and add examples?

**Comment by philss** [comment]
@jjcarstens implemented in #93.

Closing the issue.  I will take a look in another pseudo-classes before releasing a new version.
I'm gonna let you know here.
